vault/diagnose/os_common.go

```diff
@@ -38,7 +37,21 @@ partLoop:
 				SpotWarn(ctx, testName, fmt.Sprintf(partition.Mountpoint+" is %.2f percent full.", usage.UsedPercent),
 					Advice("It is recommended to have more than five percent of the partition free."))
 			} else if usage.Free < 1<<30 {
-				SpotWarn(ctx, testName, fmt.Sprintf(partition.Mountpoint+" has %s free.", humanize.Bytes(usage.Free)),
+				quantity := usage.Free
+				suffix := " bytes"
+				switch {
+				case quantity > 1024*1024*1024:
+					quantity /= 1024 * 1024 * 1024
+					suffix = "GB"
+				case quantity > 1024*1024:
+					quantity /= 1024 * 1024
+					suffix = "MB"
+				case quantity > 1024:
+					quantity /= 1024
+					suffix = "KB"
+				}
+
+				SpotWarn(ctx, testName, fmt.Sprintf(partition.Mountpoint+" has %s%s free.", quantity, suffix),
 					Advice("It is recommended to have at least 1 GB of space free per partition."))
```

Comment:
```suggestion
				suffix := "bytes"
				switch {
				case quantity > 1024*1024*1024:
					quantity /= 1024 * 1024 * 1024
					suffix = "GB"
				case quantity > 1024*1024:
					quantity /= 1024 * 1024
					suffix = "MB"
				case quantity > 1024:
					quantity /= 1024
					suffix = "KB"
				}

				SpotWarn(ctx, testName, fmt.Sprintf(partition.Mountpoint+" has %d %s free.", quantity, suffix),
					Advice("It is recommended to have at least 1 GB of space free per partition."))
```

Two things:
- `quantity` should be formatted using `%d`  otherwise you get something like "/ has %!s(uint64=42)GB free."
- Was it on purpose to put a space before "bytes", but not before the other units? My suggestion removes it from `" bytes"` and adds a space into the format string instead.

Reply:
Yeah, it seems that while "100MB" is common and "100 bytes" is obviously correct, "100 MB" is _more correct_ in that case, too.